### ISSUE TITLE
[BUG] Getting the error while running on my local jupyter notebook

### ISSUE THREAD

**Issue by bindu-0107** [Arize-ai/phoenix#1120]
**Getting the below error while running px.launch_app(conv_ds)**
Traceback (most recent call last):
  File "/home/bindupriyakamma/anaconda3/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/bindupriyakamma/anaconda3/lib/python3.9/threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bindupriyakamma/anaconda3/lib/python3.9/site-packages/uvicorn/main.py", line 381, in run
    loop = asyncio.get_event_loop()
  File "/home/bindupriyakamma/anaconda3/lib/python3.9/asyncio/events.py", line 642, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'Thread-8'.
ERROR:    💥 Phoenix failed to start. Please try again or file an issue with us at https://github.com/Arize-ai/phoenix

**Comment by mikeldking** [comment]
Hey @bindu-0107 thanks so much for trying phoenix and for filing a big report! From the stack trace it's a bit difficult to decipher the problem. Could you provide us a few details so we can reproduce as quickly as possible?

- Any notebook details you may have (how you setup the data)
- The version of Phoenix and other packages e.g. pip freeze output
- Your OS
- Is it reproducible repeatedly?

Thanks again!

**Comment by bindu-0107** [comment]
Hi @mikeldking 

- I was using to use the notebook - [find_cluster_export_and_explore_with_gpt.ipynb](https://github.com/Arize-ai/phoenix/blob/main/tutorials/find_cluster_export_and_explore_with_gpt.ipynb)

- The version I am using was arize-phoenix==0.0.32

- The issue is reporducing everytime in my local

**Comment by RogerHYang** [comment]
Hi @bindu-0107. I'm not able to reproduce this issue. Can you tell us more about your OS, e.g. Windows, Mac or Linux?

**Comment by RogerHYang** [comment]
@bindu-0107 In a fresh Jupyter Notebook, or with just the Python console, are you able to run the following?
```python
import phoenix as px
import pandas as pd
px.launch_app(px.Dataset(pd.DataFrame(), px.Schema()))
```

**Comment by bindu-0107** [comment]
@RogerHYang seems the code is working now after adding the parameter can you pls let me know what is the issue?

**Comment by RogerHYang** [comment]
Not really sure, first time encountering this. The extra parameter launches the app in a background process so it would be separate from the current thread. The downside is that the data would need to be serialized and passed to the second process.

**Comment by bindu-0107** [comment]
ok thanks @RogerHYang  for the resolution.
Can I know if weaviate support is available for phoenix and can I know if the actual embeddings are still being loaded after the UMAP creation also.(I want to check the memory utilisation of the pheonix for my use case is there any way to do it).

**Comment by mikeldking** [comment]
@bindu-0107 thanks for filing this bug. I think we are still puzzled as to why you cannot run phoenix in a thread but are glad you are un-blocked. A few questions I have (also happy to chat on slack or on a call at some point to get your perspective, LMK)

> Can I know if weaviate support is available for phoenix 
Phoenix is not really opinionated about the vector store you use - as long as you can materialize some of your vector store data in a [corpus dataset](https://docs.arize.com/phoenix/how-to/define-your-schema/corpus-data) you should be able to visualize it. What might be trickier is mapping the query embeddings to the documents. This by default works with our `OpenInferenceCallback` in LlamaIndex and we are working with LangChain to make this plug and play. Hope that helps.

>  know if the actual embeddings are still being loaded after the UMAP creation
Can you clarify this question a bit? UMAP is there to just visualize the embeddings you pass in. If your question is whether the application continues to monitor the app as you run it - the answer is unfortunately no. The app treats the datasets you pass in as immutable samples that you are testing. For continuous monitoring, we definitely recommend checking out the arize platform with continuously monitors your production data. It's free to start and free for most small apps.

> I want to check the memory utilisation of the pheonix 
Phoenix is light-weight enough to run in a notebook so it just depends on the amount of data you send in. But it's not meant to be a long running service for now so we think it's best to just stand it up when you need it. Hope that makes sense.

**Comment by bindu-0107** [comment]
@mikeldking I am getting error while running my poetry.

`
  • Installing hdbscan (0.8.33): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-cpython-310
  creating build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/validity.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/__init__.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/hdbscan_.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/plots.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/robust_single_linkage_.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/flat.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  copying hdbscan/prediction.py -> build/lib.linux-x86_64-cpython-310/hdbscan
  creating build/lib.linux-x86_64-cpython-310/hdbscan/tests
  copying hdbscan/tests/test_prediction_utils.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
  copying hdbscan/tests/__init__.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
  copying hdbscan/tests/test_flat.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
  copying hdbscan/tests/test_rsl.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
  copying hdbscan/tests/test_hdbscan.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
  running build_ext
  cythoning hdbscan/_hdbscan_tree.pyx to hdbscan/_hdbscan_tree.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_tree.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning hdbscan/_hdbscan_linkage.pyx to hdbscan/_hdbscan_linkage.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_linkage.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning hdbscan/_hdbscan_boruvka.pyx to hdbscan/_hdbscan_boruvka.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_boruvka.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning hdbscan/_hdbscan_reachability.pyx to hdbscan/_hdbscan_reachability.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_reachability.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning hdbscan/_prediction_utils.pyx to hdbscan/_prediction_utils.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_prediction_utils.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  cythoning hdbscan/dist_metrics.pyx to hdbscan/dist_metrics.c
  /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/dist_metrics.pxd
    tree = Parsing.p_module(s, pxd, full_module_name)
  building 'hdbscan._hdbscan_tree' extension
  creating build/temp.linux-x86_64-cpython-310
  creating build/temp.linux-x86_64-cpython-310/hdbscan
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/tmpyz2nediq/.venv/include -I/usr/include/python3.10 -I/tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/numpy/core/include -c hdbscan/_hdbscan_tree.c -o build/temp.linux-x86_64-cpython-310/hdbscan/_hdbscan_tree.o
  hdbscan/_hdbscan_tree.c:6:10: fatal error: Python.h: No such file or directory
      6 | #include "Python.h"
        |          ^~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
  

  at ~/anaconda3/lib/python3.9/site-packages/poetry/installation/chef.py:147 in _prepare
      143│ 
      144│                 error = ChefBuildError("\n\n".join(message_parts))
      145│ 
      146│             if error is not None:
    → 147│                 raise error from None
      148│ 
      149│             return path
      150│ 
      151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with hdbscan (0.8.33) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "hdbscan (==0.8.33)"'.
`


can you pls help me in resolving this hdbscan error i tried so many things but it didnt work

**Comment by mikeldking** [comment]
> @mikeldking I am getting error while running my poetry.
> 
> 
> 
> `
> 
>   • Installing hdbscan (0.8.33): Failed
> 
> 
> 
>   ChefBuildError
> 
> 
> 
>   Backend subprocess exited when trying to invoke build_wheel
> 
>   
> 
>   running bdist_wheel
> 
>   running build
> 
>   running build_py
> 
>   creating build
> 
>   creating build/lib.linux-x86_64-cpython-310
> 
>   creating build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/validity.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/__init__.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/hdbscan_.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/plots.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/robust_single_linkage_.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/flat.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   copying hdbscan/prediction.py -> build/lib.linux-x86_64-cpython-310/hdbscan
> 
>   creating build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   copying hdbscan/tests/test_prediction_utils.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   copying hdbscan/tests/__init__.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   copying hdbscan/tests/test_flat.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   copying hdbscan/tests/test_rsl.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   copying hdbscan/tests/test_hdbscan.py -> build/lib.linux-x86_64-cpython-310/hdbscan/tests
> 
>   running build_ext
> 
>   cythoning hdbscan/_hdbscan_tree.pyx to hdbscan/_hdbscan_tree.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_tree.pyx
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   cythoning hdbscan/_hdbscan_linkage.pyx to hdbscan/_hdbscan_linkage.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_linkage.pyx
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   cythoning hdbscan/_hdbscan_boruvka.pyx to hdbscan/_hdbscan_boruvka.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_boruvka.pyx
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   cythoning hdbscan/_hdbscan_reachability.pyx to hdbscan/_hdbscan_reachability.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_hdbscan_reachability.pyx
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   cythoning hdbscan/_prediction_utils.pyx to hdbscan/_prediction_utils.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/_prediction_utils.pyx
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   cythoning hdbscan/dist_metrics.pyx to hdbscan/dist_metrics.c
> 
>   /tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/tmpuypx23go/hdbscan-0.8.33/hdbscan/dist_metrics.pxd
> 
>     tree = Parsing.p_module(s, pxd, full_module_name)
> 
>   building 'hdbscan._hdbscan_tree' extension
> 
>   creating build/temp.linux-x86_64-cpython-310
> 
>   creating build/temp.linux-x86_64-cpython-310/hdbscan
> 
>   x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/tmpyz2nediq/.venv/include -I/usr/include/python3.10 -I/tmp/tmpyz2nediq/.venv/lib/python3.10/site-packages/numpy/core/include -c hdbscan/_hdbscan_tree.c -o build/temp.linux-x86_64-cpython-310/hdbscan/_hdbscan_tree.o
> 
>   hdbscan/_hdbscan_tree.c:6:10: fatal error: Python.h: No such file or directory
> 
>       6 | #include "Python.h"
> 
>         |          ^~~~~~~~~~
> 
>   compilation terminated.
> 
>   error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
> 
>   
> 
> 
> 
>   at ~/anaconda3/lib/python3.9/site-packages/poetry/installation/chef.py:147 in _prepare
> 
>       143│ 
> 
>       144│                 error = ChefBuildError("\n\n".join(message_parts))
> 
>       145│ 
> 
>       146│             if error is not None:
> 
>     → 147│                 raise error from None
> 
>       148│ 
> 
>       149│             return path
> 
>       150│ 
> 
>       151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:
> 
> 
> 
> Note: This error originates from the build backend, and is likely not a problem with poetry but with hdbscan (0.8.33) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "hdbscan (==0.8.33)"'.
> 
> `
> 
> 
> 
> 
> 
> can you pls help me in resolving this hdbscan error i tried so many things but it didnt work
@bindu-0107

Unfortunately I don't maintain HDBSCAN so I can't be of much assistance:( I know the maintainers though and they are super friendly! Definitely reach out to them and I'm sure they will help you out.

**Comment by giaosudau** [comment]
```
Failed to build hdbscan
ERROR: Could not build wheels for hdbscan, which is required to install pyproject.toml-based projects

[notice] A new release of pip available: 22.3.1 -> 23.3.2
[notice] To update, run: pip install --upgrade pip

```
Which version of hdbscan I can use?

**Comment by mikeldking** [comment]
@giaosudau - we haven't looked into this issue in a bit. There was a major change in cython that caused more issues. Our package should work for HDBCAN in this range ( "hdbscan>=0.8.33, <1.0.0"). Could you try installing HDBSCAN on it's own? If it doesn't work the HDBSCAN authors are super helpful.

Also can you tell us what your setup is? E.g. OS, python version, pip version etc?

Thanks for trying out the package! Happy to help in any way.
